### PR TITLE
Add publictests test checker to druntime (reboot of #2087)

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Checking public examples is always worthwhile IMHO.

- [x] dlang/tools#402
- [x] dlang-community/libdparse#391
- [x] dlang-community/libdparse#398
- [x] dlang-community/libdparse#401
- [x] dlang-community/libdparse#402
- [x] dlang-community/libdparse#404 (optional)
- [x] dlang/tools#403
- [x] Update libdparse used by `test_extractor.d` to include latest bugfixes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,3 +25,6 @@ jobs:
       - run:
           command: ./.circleci/run.sh betterc
           name: Run @betterC tests
+      - run:
+          command: ./.circleci/run.sh publictests
+          name: Run public unittests
```
Comment (wilzbach) [inline .circleci/config.yml:30]: Just as an FYI this won't apply to all existing PRs. They need to be rebased for them to have this test.

**Comment by MoonlightSentinel** [comment]
Rebased to retrigger the CI

**Comment by MoonlightSentinel** [comment]
Not sure why Win32 fails on Azure & auto-tester fails reliably as this doesn't touch windows specific stuff. Anybody got an idea?

**Comment by MoonlightSentinel** [comment]
Nevermind, it's an Optlink error that opens a dialog ... 

```
Unexpected OPTLINK Termination at EIP=004211CB
```

**Comment by MoonlightSentinel** [comment]
Using `fabs` instead of an explicit expressions seems to work.

**Pull request merged**